surfactant/cmd/cli.py

```diff
@@ -17,3 +62,96 @@ def edit(sbom):
 @click.command("add")
 def add(sbom):
     "CLI command to add specific entry(s) to a supplied SBOM"
+
+
+class cli_find:
+    """
+    A class that implements the surfactant cli find functionality
+
+    Attributes:
+    match_functions     A dictionary of functions that provide matching functionality for given SBOM fields (i.e. uuid, sha256, installpath, etc)
+    sbom                A internal record of sbom entries the class adds to as it finds more matches.
+    """
+
+    match_functions: dict
+    camel_case_conversions: dict
+    sbom: SBOM
+
+    def __init__(self):
+        """Initializes the cli_find class"""
+        self.match_functions = {
+            int: self.match_single_value,
+            str: self.match_single_value,
+            list: self.match_array_value,
+        }
+        self.camel_case_conversions = {
+            "uuid": "UUID",
+            "filename": "fileName",
+            "containerpath": "containerPath",
+            "installpath": "installPath",
+            "capturetime": "captureTime",
+            "relationshipassertion": "relationshipAssertion",
+        }
+        self.sbom = SBOM()
+
+    def execute(self, input_sbom: SBOM, **kwargs):
+        """Executes the main functionality of the cli_find class
+        param: input_sbom   The sbom to find matches within
+        param: kwargs:      Dictionary of key/value pairs indicating what features to match on
+        """
+        for sw in input_sbom.software:
+            match = True
+            for k, v in kwargs.items():
+                if k == "file":
+                    entry_value = sw.sha256
+                    v, _, _ = self._calculate_hashes(v, sha256=True)
+                else:
+                    entry_value = vars(sw)[
+                        self.camel_case_conversions[k] if k in self.camel_case_conversions else k
+                    ]
+                if not self.match_functions[type(entry_value)](entry_value, v):
```

Comment:
What happens if the entry value decoded from the input SBOM file is a different type?

This is one place that I think the Python 3.10 match..case syntax would be nice to use; but I'm not quite ready to drop support for Python 3.8 and 3.9 yet.

Reply:
Since we are matching types from the SBOM class, we should not be getting float, dict or tuples. I've added a handler case for None, but I can add the rest of the types and match them with a generic non implementation function

Reply:
Updated dic to include values for float, dict, none, and tuple. If a user supplies an SBOM that contains a key with value of that type, it will call the match_none_or_unhandled function